app/controllers/concerns/authentication.rb

```diff
@@ -328,10 +328,12 @@ def request_http_basic_authentication(realm = 'keygen', message = nil)
     raise Keygen::Error::UnauthorizedError.new(code: 'TOKEN_INVALID')
   end
 
-  # we only support cookie authn from same- or sub-origins (i.e. subdomains)
+  # NOTE(ezekg) we only support cookie authn from same-site origins (e.g. subdomains) to prevent CSRF
+  #
+  #             see: https://scotthelme.co.uk/csrf-is-dead/
   def has_cookie_credentials?
     (request.origin.nil? || request.origin.ends_with?(Keygen::DOMAIN)) &&
```

Comment:
This introduces a CRSF attack vector from subdomains out of our control, like `status.keygen.sh`. We should either explicitly whitelist the domains that can use cookie authn, or move these third-party subdomains onto a separate domain, e.g. `keygenstatus.com`. The former is the safest route.